cucumber/api/features/step_definitions/authz_steps.rb

```diff
@@ -82,7 +82,7 @@
 end
 
 Given(/^I create a binary secret value?$/) do
-  @value = Random.new.bytes(16)
+  @value = Random.new.bytes(16).force_encoding(Encoding::UTF_8)
```

Comment:
I don't think this follows the original intent of this step. The value of a Secret should be able to be *any* binary data, not just data that's representable as UTF-8 (see http://haacked.com/archive/2012/01/30/hazards-of-converting-binary-data-to-a-string.aspx/).

Could be that the problem is actually in the comparison step?